cmd/hostcontroller/main.go

```diff
@@ -107,12 +110,21 @@ func main() {
 		SecureServing: args.secureMetrics,
 		TLSOpts:       args.tlsOpts,
 	}
-
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:                 scheme,
 		HealthProbeBindAddress: args.probeAddr,
-		Cache:                  cache.Options{},
 		Metrics:                metricsServerOptions,
+		// Restrict client cache/informer to events for the node running this pod.
+		// On large clusters, not doing so can overload the API server for daemonsets
+		// since nodes receive frequent updates in some environments.
+		Cache: cache.Options{
+			ByObject: map[client.Object]cache.ByObject{
+				&corev1.Node{}: {
+					Field: fields.Set{"metadata.name": args.nodeName}.AsSelector(),
+				},
+				// TODO: We should do the same for pods since it's a massive resource
```

Comment:
yes, whatever we listen to (pods, nodes, whatever), we should only listen to instances scheduled in this particular node. 

Is this implemented in a follow-up commit ?